src/main/java/org/elasticsearch/transport/couchbase/capi/CouchbaseCAPITransportImpl.java

```diff
@@ -119,7 +143,7 @@ public CouchbaseCAPITransportImpl(Settings settings, RestController restControll
         this.typeSelector.configure(settings);
         logger.info("Couchbase transport is using type selector: {}", typeSelector.getClass().getCanonicalName());
 
-        Class<? extends ParentSelector> parentSelectorClass = settings.<ParentSelector>getAsClass("couchbase.parentSelector", DefaultParentSelector.class);
+        Class<? extends ParentSelector> parentSelectorClass = DefaultParentSelector.class;
```

Comment:
@Branor In ES 2.0 and 2.x, there is no longer support for loading class names from Strings.  Settings objects no longer have a visible ClassLoader either.  The following code I think works around this problem, but I don't know if the "local" ClassLoader to the Impl class is guaranteed to be the right one:

This private method is added to the Impl class:

    @SuppressWarnings({"unchecked"})
    private <T> Class<? extends T> getAsClass(
      String className, Class<? extends T> defaultClazz) {
      String sValue = className;
      if (sValue == null) {
         return defaultClazz;
      }
      try {
         return (Class<? extends T>) this.getClass().getClassLoader().loadClass(sValue);
      } catch (ClassNotFoundException e) {
         throw new NoClassSettingsException("Failed to load class setting [" 
            + className + "] with value [" + sValue + "]", e);
      }
    }

The line in question is changed to:

    Class<? extends TypeSelector> typeSelectorClass = this.getAsClass(settings.get("couchbase.typeSelector"), DefaultTypeSelector.class);

I'll commit these changes to the PR if you think they are a suitable alternative to calling the now missing Settings.getAsClass method.